ui.apps/src/main/content/jcr_root/apps/asset-share-commons/clientlibs/clientlib-site/js/storage.js

```diff
@@ -0,0 +1,262 @@
+    function _setUserProfile(profile) {
+        var storage;
+
+        if(typeof profile === 'undefined' || profile.type !== 'user') {
+            return;
+        }
+
+        storage = _getLocalStorage();
+
+        //set current user
+        currentUserId = profile.authorizableId_xss;
+        
+        //update existing user
+        if(storage["users"] && storage["users"][currentUserId]) {
+            storage["users"][currentUserId].profile = profile;
+        } else {
+            //add new user
+            storage["users"] = storage["users"] || {};
+            storage["users"][currentUserId] = {profile: profile};
+        }
+
+        _updateLocalStorage(storage);
+    
+    }
+
+    /* adds an asset path to the cart and persists to local storage */
+    function addCartAsset(assetPath) {
+        var storage = _getLocalStorage(),
+            assets;
+
+        if(storage["users"] && storage["users"][currentUserId]) {
+
+            storage["users"][currentUserId]["cart"] = storage["users"][currentUserId]["cart"] || {};
+            assets = storage["users"][currentUserId]["cart"]["assets"] || [];
+
+            if(assets.indexOf(assetPath) < 0) {
+                assets.push(assetPath);
+                storage["users"][currentUserId]["cart"]["assets"] = assets;
+                _updateLocalStorage(storage);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /* removes an asset path from the cart and persists to local storage */
+    function removeCartAsset(assetPath) {
+        var storage = _getLocalStorage(),
+            assets,
+            index;
+
+        if(storage["users"] && storage["users"][currentUserId] && storage["users"][currentUserId].cart) {
+            assets = storage["users"][currentUserId]["cart"]["assets"] || [];
+
+            index = assets.indexOf(assetPath);
+            if( index >= 0) {
+                assets.splice(index,1);
+                storage["users"][currentUserId]["cart"]["assets"] = assets;
+                _updateLocalStorage(storage);
+                return true;
+            }
+        }
+        return false;
+    }
+
+     /* clears the cart and sets it to an empty array */
+     function clearCartAssets() {
+        var storage = _getLocalStorage();
+
+        if(storage["users"] && storage["users"][currentUserId]) {
+            storage["users"][currentUserId]["cart"] = storage["users"][currentUserId]["cart"] || {};
+            storage["users"][currentUserId]["cart"]["assets"] = [];
+
+            _updateLocalStorage(storage);
+            return true;
+        }
+
+        return false;
+    }
+
+    /* Gets the contents of the cart Array, if not set an empty array is returned */
+    function getCartAssets() {
+        var storage,
+            paths = [];
+
+        storage = _getLocalStorage();
+        if(storage["users"] && storage["users"][currentUserId] && storage["users"][currentUserId].cart) {
```

Comment:
Sees like we do this type of object check alot... 

Could this be pulled into a method? And/or used in conjunction with the optional syntax `storage?.users?.[currentUserId]?.cart` ? (i would think our babel would handle it?)

Reply:
ah i actually wasn't familar with the `?` syntax. I can play with it. I don't believe this JS is being run through babel since its in the clientlib directly...